flake.nix

```diff
@@ -1,47 +1,93 @@
+                allowUnsupportedSystem = true;
+                allowBroken = true;
+              };
+            };
+
         in
-        with pkgs;
         {
-          devShells.default = mkShell {
-            buildInputs = [
-              just
-              xcode-install
+          devShells.default = pkgs.mkShell {
+            buildInputs = with pkgs; [
+              bashInteractive
+              colima
+              create-project
+              devcontainers
               direnv
-              ripgrep
-              python3
-              nodejs_20
               docker
               docker-compose
+              docker-credential-helpers
+              gum
+              just
+              nixpkgs-fmt
+              nodejs_20
               nodejs_20.pkgs.pnpm
+              python3
+              ripgrep
+              xcode-install
             ];
-            shellHook = ''
-            if [[ ! -e local-dev ]]; then
-              git clone git@github.com:scorbettUM/local-dev.git
-              echo 'eval "$(direnv hook zsh)"' | sudo tee -a $HOME/.zshrc > /dev/null
-
-              touch $HOME/.envrc
-              echo 'use flake "github.com:scorbettUM/local-dev"' | sudo tee $HOME/.envrc > /dev/null
+            shellHook =
+              let
+                sourceEnv = f: ''
+                  if [[ -f ${f} ]]; then
+                    echo 1>&2 'sourcing env variables from ${f}'
+                    set -a
+                    . ${f}
+                    set +a
+                  fi
+                '';
+              in
 
-              cp local-dev/justfile $HOME/justfile
-              
-              rm -rf local-dev
-            fi
+              ''
+                gum style --border double \
+                  --align center \
+                  --width 50 \
+                  --margin "1 2" \
+                  --padding "2 4" \
+                  'Welcome to local dev!' \
+                  'I hope you enjoy your stay! If you need help, just holler'
+                ${sourceEnv ".env"}
+                ${sourceEnv ".env.local"}
 
-            sudo npm install -g @devcontainers/cli
-            '';
+                [[ -f justfile  ]] && command -v just >/dev/null 2>&1 && just --list --unsorted
+              '';
+          };
+        }) // {
+      overlays.default = final: prev: {
+        devcontainers = prev.mkYarnPackage {
+          name = "devcontainer";
+          src = prev.fetchFromGitHub {
+            owner = "devcontainers";
+            repo = "cli";
+            rev = "v0.56.1";
+            hash = "sha256-Q9AYPUJPcPQzSYCQyU0PTz4Cgn16E541IMxT5ofrmHE=";
           };
-        }
-      );
-}
\ No newline at end of file
+          buildPhase = ''
+            export HOME=$(mktemp -d)
+            yarn --offline compile
+          '';
+          dontStrip = true;
+        };
+        create-project = prev.writers.writePython3Bin "create-project"
+          {
+            libraries = [ ];
```

Comment:
If you want to, eg, add any library in nixpkgs here, you can.

```nix
libraries = with prev.python3Packages; [ pandas numpy boto3 hypothesis  ]; # etc etc
```

and then just "import" then in `create-project.py` as if you had them "pip installed" normally.